extension/content_script.js

```diff
@@ -635,21 +640,64 @@ function toggleExtensionVisibility() {
 
 function setupListenerForOnOff() {
 	chrome.runtime.onMessage.addListener(	
-		function(request, sender, sendResponse) {	
-			if (request.command === "toggleUI") {		
+		function(request, sender, sendResponse) {
+			function toggleApp() {
 				if (doc === null) {
 					// Make sure these load after animation
-					$("#loadingIcon").show(100, function() {
+					$("#loadingIcon").show(500, function() {
 						preprocessPage();
 						toggleExtensionVisibility();
 					});	
 				}
 				else { toggleExtensionVisibility(); }
+			}			
+			if (request.command === "toggleUI") {
+				toggleApp();
+			}	
+			// TODO: This can be perhaps refactored. For now, it's an easy fir
+			else if (request.command === "toggleUITutorial") {
+				if (settings.getAppStatus()) {
+					return; // If app status is on, do nothing
+				} 
+				toggleApp();
 			}	
+			
 		}	
 	);
 };
 
+function setupTutorial() {
+	// TODO: Move these into separate file
+	let tutorialPopupHtml = `
+		<div id="tutorialPopupContainer">
+			<div class="popupText">Click the<span id="optionsButtonTutorial">${window.gearLogo}</span>for instructions</div>
+			<div class="popupCheckmark"></div>
+		</div>
+	`;
+	chrome.runtime.onMessage.addListener(	
+		function(request, sender, sendResponse) {	
+			if (request.command === "startTutorial") {	
+				$(tutorialPopupHtml).insertAfter($("body").children().first());
+				let tutorialPopup = $("#tutorialPopupContainer");
+				$(".popupCheckmark").css("background-image", "url("+popupCheckmarkUrl+")").click(function() {
+					$("#optionsButton").click();
+					tutorialPopup.remove()
+				});
+				$("#optionsButtonTutorial").click(function() {
```

Comment:
I wonder how the UI will look like if user just clicks on the gear icon directly. -- should autoclose instruction probably?
If that case is not handled, please leave a todo.

Reply:
If you mean the normal gear (not the one in instructions), no bc wasn't sure what we'd want anyway, but good point, adding todo

Reply:
yeah sorry i meant normal gear, because I could see myself doing that.
Instead of "click tick", the resolving action would be "click gear"
todo sg.

Reply:
** and maybe tick isn't even necessary.
tick clicked -> instruction pops up 
vs
gear clicked -> instruction pops up

might as well have them learn the real flow = second flow

Reply:
Good point! Have tutorial progress when they do things, not by clicking on checkmark